youtube_dl/downloader/__init__.py

```diff
@@ -30,6 +30,10 @@
 def get_suitable_downloader(info_dict, params={}):
     """Get the downloader class that can handle the info dict."""
     protocol = determine_protocol(info_dict)
+    if protocol == 'm3u8' and params.get('hls_prefer_native') is True:
+        protocol = 'm3u8_native'
+    elif protocol == 'm3u8_native' and params.get('hls_prefer_native') is False:
+        protocol = 'm3u8'
```

Comment:
`protocol` should not be touched. `protocol` is protocol selected according to extracted info dictionary not according to provided command line options' configuration.